.github/workflows/label_prs.yaml

```diff
@@ -55,17 +71,35 @@ jobs:
           PR: ${{ github.event.pull_request.html_url }}
 
       - id: infrastructure
-        if: steps.filter.outputs.docs == 'true'
+        if: steps.filter.outputs.infrastructure == 'true'
         run: |
           gh pr edit "${PR}" --add-label "infrastructure"
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           PR: ${{ github.event.pull_request.html_url }}
 
+      - id: proposal
+        if: steps.filter.outputs.proposal == 'true'
+        run: |
+          gh pr edit "${PR}" --add-label "proposal"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PR: ${{ github.event.pull_request.html_url }}
```

Comment:
Proposals should be getting labels from the proposal_labeled and proposal_ready flows, are you seeing issues with them? I'm wary of having multiple actions trying to set labels.

Reply:
So, I believe that `proposal_labeled.yaml` only handles keeping updates between the draft, rfc, accepted, and declined labels consistent. And `proposal_ready.yaml` only handles a PR that has some proposal label getting marked as `ready_for_review`, and triggering the `proposal rfc` label for that. A human has to apply the `proposal` (or `proposal draft`) label first.

And those workflows don't seem the right place to do this `proposal` labeling as we really want to trigger on the *paths* here and more events. So I think this is a reasonably clean breakdown of logic based on the events that trigger workflows.

I can try to merge all the proposal label management into a single workflow if you want, but we already have 3 actions setting labels, and so I don't think this PR is making things much worse there.

Reply:
So, new_proposal.py adds the proposal label. Should PRs that don't use the new_proposal script that add files here generally be marked as proposals? That will probably include non-proposal things.

Reply:
As discussed, hopefully this actually matches what we want and gives coverage for proposals that aren't created with the script.

I've also added a comment to highlight how it is different.